android-components/components/browser/state/src/main/java/mozilla/components/browser/state/engine/middleware/TranslationsMiddleware.kt

```diff
@@ -41,6 +47,25 @@ class TranslationsMiddleware(
                     requestSupportedLanguages(context, action.tabId)
                 }
             }
+
+            is TranslationsAction.TranslateDataRequestedAction -> {
+                when (action.operation) {
+                    TranslationOperation.TRANSLATE -> {
+                        // No-op, not a data operation
+                    }
+                    TranslationOperation.RESTORE -> {
+                        // No-op, not a data operation
+                    }
+                    TranslationOperation.FETCH_LANGUAGES -> {
+                        // Bug 1877205
```

Comment:
I'm thinking [Bug 1877205](https://bugzilla.mozilla.org/show_bug.cgi?id=1877205) will simply move `requestSupportedLanguages(context, action.tabId)` here and fix the test.

Then in [Bug 1873672](https://bugzilla.mozilla.org/show_bug.cgi?id=1873672) will figure out where `TranslateDataRequestedAction` should be called from.